providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/FlagStore.java

```diff
@@ -31,6 +32,9 @@ public class FlagStore implements Storage {
     private final BlockingQueue<StorageState> stateBlockingQueue = new LinkedBlockingQueue<>(1);
     private final Map<String, FeatureFlag> flags = new HashMap<>();
 
+    @Getter
```

Comment:
This is another place where we have some state which presents potential bugs and concurrency issues.

Similar to what you did with the `stateConsumer` function, I think you need to modify the `StorageState` object not to be just an enum, but a field which contains and enum AND a list of changed flags. Then, when, we can offer both to the queue so we can avoid this state.

I hope that makes sense! Other than that, everything looks great here. Thanks for all your efforts!

Reply:
understood will implement this 👍

Reply:
@toddbaert have implemented this please help with review